core/controllers/base.py

```diff
@@ -278,6 +294,11 @@ def delete(self, *args):  # pylint: disable=unused-argument
         raise self.PageNotFoundException
 
     def render_json(self, values):
+        """Prepares JSON response to be sent to the client.
+
+        Args:
+            values: dict. The key-value pairs to encode in the JSON response.
+        """
         self.response.content_type = 'application/javascript; charset=utf-8'
         self.response.headers['Content-Disposition'] = (
             'attachment; filename="oppia-attachment.txt"')
```

Comment:
This is independent, and has no relevance to the `filename` arg that was passed in before.